src/jarabe/model/shell.py

```diff
@@ -723,23 +723,32 @@ def notify_launch(self, activity_id, service_name):
             GLib.source_remove(self._launch_timers[activity_id])
             del self._launch_timers[activity_id]
 
-        timer = GLib.timeout_add_seconds(90, self._check_activity_launched,
+        timer = GLib.timeout_add_seconds(10, self._check_activity_launched,
```

Comment:
The commit that introduced this is https://github.com/sugarlabs/sugar/commit/6528ba50dfcf2fd59f22812387975beacc1f3fdc, did you check to confirm if the issue still exists after your change?